### PR TITLE
Feat: vela status --tree

### PR DESCRIPTION
**Pull request by Somefive**
Signed-off-by: Somefive <yd219913@alibaba-inc.com>

### Description of your changes

`vela status --tree` shows the hierarchical structure of applied resources.
- It will show all the resources the application currently holds, including outdated ones. (This is done by retrieving all the resources from ResourceTrackers.)
- With the `--detail` option, it will fetch the resources dynamically, and display the detail information of the resources.
- With the `--detail-format` option, it can display resources in several different ways.

Examples
- vela status --tree
image
- vela status --tree --detail 
image
- vela status --tree --detail --detail-format=wide
image
- vela status --tree --detail --detail-format=table
image
- vela status --tree --detail --detail-format=list
image
- vela status --tree --detail --detail-format=raw
image

A large example:
image

A example of application with un-deployed clusters.
image

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

More future work for this command:
1. format: the overflow problem of details is not handled. So if the detail line is too long, it will break the tree structure.
2. alternative structure: this view is a pure resource view, without component/trait. Alternative view should be component/trait in replace of namespace. One difficulty here is the belonging relationship between component and namespace is undetermined. A component might have resources across namespaces. A namespace can also hold multiple components.
3. future `vela top` command could make the display of details interactively changing.

### CODE REVIEW

**Review by zzxwill — APPROVED** [review]
Great job!

**Comment by Somefive** [comment]
> Great job!

I will add some test fix later. Thanks for the review. :)

File: pkg/policy/topology.go [hunk 1]
```diff
@@ -33,3 +40,57 @@ func GetClusterLabelSelectorInTopology(topology *v1alpha1.TopologyPolicySpec) ma
...
+						return nil, err
+					}
+				}
+			case clusterLabelSelector != nil:
+				clusters, err := multicluster.FindVirtualClustersByLabels(context.Background(), cli, clusterLabelSelector)
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to find clusters in topology %s", policy.Name)
+				}
+				if len(clusters) == 0 {
+					return nil, errors.Errorf("failed to find any cluster matches given labels")
```
Comment (wonderflow) [inline pkg/policy/topology.go:85]:
```suggestion
					return nil, errors.New("failed to find any cluster matches given labels")
```

File: pkg/resourcetracker/tree.go [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+	resourceName         string
+	connectClusterUp     bool
+	connectClusterDown   bool
+	connectNamespaceUp   bool
+	connectNamespaceDown bool
+	applyTime            string
+	details              string
+}
+
+func (options ResourceTreePrintOptions) loadResourceRows(currentRT *v1beta1.ResourceTracker, historyRT []*v1beta1.ResourceTracker) []*resourceRow {
```
Comment (wonderflow) [inline pkg/resourcetracker/tree.go:75]:
```suggestion
func (options *ResourceTreePrintOptions) loadResourceRows(currentRT *v1beta1.ResourceTracker, historyRT []*v1beta1.ResourceTracker) []*resourceRow {
```

**Review by wonderflow — APPROVED** [review]
LGTM great work

**Review by FogDong — APPROVED** [review]

**Pull request merged**